/* Copyright 2019 Istio Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package stats;

message PluginConfig {
  // The following settings should be rarely used.
  // Enable debug for this filter.
  bool debug = 1;

  // maximum size of the peer metadata cache.
  // A long lived proxy that connects with many transient peers can build up a
  // large cache.
  int32 max_peer_cache_size = 2;

  // Stats api squashes dimensions in a single string.
  // The squashed string is parsed at prometheus scrape time to recover
  // dimensions. The following 2 fields set the field and value separators {key:
  // value} -->  key{value_separator}value{field_separator}
  string field_separator = 3;  // default: ";;"
  string value_separator = 4;  // default: "=="
}
